examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairingCommand.kt

```diff
@@ -0,0 +1,253 @@
+      PairingNetworkType.NONE -> {}
+      PairingNetworkType.WIFI -> {
+        addArgument("ssid", SSID, null, false)
+        addArgument("password", password, null, false)
+      }
+
+      PairingNetworkType.THREAD -> addArgument("operationalDataset", operationalDataset, null, false)
+    }
+    when (mode) {
+      PairingModeType.NONE -> {}
+      PairingModeType.CODE, PairingModeType.CODE_PASE_ONLY -> {
+        addArgument("payload", onboardingPayload, null, false)
+        addArgument("discover-once", discoverOnce, null, false)
+        addArgument("use-only-onnetwork-discovery", useOnlyOnNetworkDiscovery, null, false)
+      }
+
+      PairingModeType.ADDRESS_PASE_ONLY -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+
+      PairingModeType.BLE -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("discriminator", 0.toShort(), 4096.toShort(), discriminator, null, false)
+      }
+
+      PairingModeType.ON_NETWORK -> addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+      PairingModeType.SOFT_AP -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("discriminator", 0.toShort(), 4096.toShort(), discriminator, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+
+      PairingModeType.ALREADY_DISCOVERED -> {
+        addArgument("setup-pin-code", 0, 134217727, setupPINCode, null, false)
+        addArgument("device-remote-ip", remoteAddr, false)
+        addArgument("device-remote-port", 0.toShort(), Short.MAX_VALUE, remotePort, null, false)
+      }
+    }
+    when (filterType) {
+      DiscoveryFilterType.NONE -> {}
+      DiscoveryFilterType.SHORT_DISCRIMINATOR -> addArgument(
+        "discriminator",
+        0.toShort(),
+        4096.toShort(),
+        discriminator,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.LONG_DISCRIMINATOR -> addArgument(
+        "discriminator",
+        0.toShort(),
+        4096.toShort(),
+        discriminator,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.VENDOR_ID -> addArgument(
+        "vendor-id",
+        0,
+        Long.MAX_VALUE,
+        discoveryFilterCode,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.COMPRESSED_FABRIC_ID -> addArgument(
+        "fabric-id",
+        0,
+        Long.MAX_VALUE,
+        discoveryFilterCode,
+        null,
+        false
+      )
+
+      DiscoveryFilterType.COMMISSIONING_MODE, DiscoveryFilterType.COMMISSIONER -> {}
+      DiscoveryFilterType.DEVICE_TYPE -> addArgument(
+        "device-type",
+        0,
+        Long.MAX_VALUE,
```

Comment:
Same for device-type -- I believe device types are 16-bit as well so could be restricted to `0xFFFFL`?